### PR TITLE
Update Reggie

### PR DESCRIPTION
**Pull request by pmengelbert**
*Tidy Execute method and found source of QueryParam bug
*Split different structs into different files

### CODE REVIEW

File: client.go [hunk 1]
```diff
@@ -45,10 +33,6 @@ type (
 	}
 )
 
-func (resp *Response) IsUnauthorized() bool {
-	return resp.StatusCode() == http.StatusUnauthorized
-}
-
 type reqConfig struct {
```
Comment (jdolitsky) [inline client.go]: can we move this type into the group of types in the top section?

File: response.go [hunk 2]
```diff
@@ -0,0 +1,62 @@
+package reggie
+
+import (
+	"encoding/json"
+	"github.com/mitchellh/mapstructure"
+	"gopkg.in/resty.v1"
+	"net/http"
+	"regexp"
```
Comment (jdolitsky) [inline response.go]: these should be reordered w/ goimports

File: response.go [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+type Response struct {
+	*resty.Response
+}
+
+func (resp *Response) IsUnauthorized() bool {
+	return resp.StatusCode() == http.StatusUnauthorized
+}
+
+func (client *Client) retryRequestWithAuth(originalRequest *Request, originalResponse *Response) (*Response, error) {
```
Comment (jdolitsky) [inline response.go]: since this method is on Client type, lets put in client.go

File: response.go [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+	if err != nil {
+		return nil, err
+	}
+
+	originalRequest.deleteQueryParams() //Otherwise they will be appended to the URL again
+	originalRequest.SetAuthToken(info.Token)
+	return originalRequest.Execute(originalRequest.Method, originalRequest.URL)
+}
+
+func parseAuthHeader(authHeaderRaw string) *authHeader {
```
Comment (jdolitsky) [inline response.go]: since this is only used by method above, should also go into client.go

File: request.go [hunk 5]
```diff
@@ -0,0 +1,35 @@
+package reggie
+
+import (
+	"regexp"
+
+	"gopkg.in/resty.v1"
+)
+
+type Request struct {
```
Comment (jdolitsky) [inline request.go]: lets switch to 
```
type (
    Request struct {
...
```

File: response.go [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+
+import (
+	"encoding/json"
+	"github.com/mitchellh/mapstructure"
+	"gopkg.in/resty.v1"
+	"net/http"
+	"regexp"
+)
+
+type Response struct {
```
Comment (jdolitsky) [inline response.go]: lets switch to 
```
type (
    Response struct {
...
```

**Review by jdolitsky — CHANGES_REQUESTED** [review]

File: client.go (hunk 1 above)
Comment (pmengelbert) [reply client.go]: yep

File: client.go [hunk 7]
```diff
@@ -3,23 +3,14 @@ package reggie
 import (
 	"encoding/json"
 	"fmt"
-	"net/http"
+	"github.com/mitchellh/mapstructure"
```
Comment (jdolitsky) [inline client.go]: one last import issue - put the third-party after the stdlib stuff

**Review by jdolitsky — CHANGES_REQUESTED** [review]

**Review by jdolitsky — APPROVED** [review]

**Pull request merged**